server/src/controllers/surveyResultController.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as express from 'express';
+import SurveyResultRequest from '../models/surveyResultRequest';
+import { SurveyResultRequestModel } from '../interfaces/surveyModel';
+import {
+  virtualVisitsDatabaseName,
+  surveyContainerName,
+  cosmosClient,
+  createDatabase,
+  createContainer
+} from '../utils/cosmosClient';
+
+export const storeSurveyResult = async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+  try {
+    const { body: requestData } = req;
+    const { sessionId, callId, acsUserId, response } = requestData;
+
+    // Validation.
+    // If any one of field does not exists in the payload the validation failed.
+    if (!(sessionId && callId && acsUserId && response)) {
+      return res.status(400).send({ error: 'Payload invalid.' });
+    }
+
+    const inputData: SurveyResultRequestModel = {
+      ...requestData,
+      dateTime: new Date()
+    };
+
+    const survey = new SurveyResultRequest(inputData);
+
+    // Create database and container.
+    await createDatabase();
+    await createContainer();
+
+    // Store survey result.
+    await cosmosClient.database(virtualVisitsDatabaseName).container(surveyContainerName).items.upsert(survey);
+
+    res.json('Store successfully.');
+  } catch (error) {
+    return next(error);
```

Comment:
[Express asynchronous error handling](https://expressjs.com/en/guide/error-handling.html#:~:text=You%20must%20catch%20errors%20that%20occur%20in%20asynchronous%20code%20invoked%20by%20route%20handlers%20or%20middleware%20and%20pass%20them%20to%20Express%20for%20processing.%20For%20example%3A)